pandas/core/window/rolling.py

```diff
@@ -1186,13 +1186,21 @@ def count(self):
         window = self._get_window()
         window = min(window, len(obj)) if not self.center else window
 
+        # We set the default value min_periods to be 0 because count method
+        # is meant to count NAs, we don't want it by default requires all
+        # values in the window to be valid to produce a valid count
+        min_periods = 0 if self.min_periods is None else self.min_periods
```

Comment:
Is this case tested? Still a little unclear why this is required; does the change you have below to just pass `min_periods` as an argument not fix the issue?

Reply:
This was not explicitly tested, but are partially covered by some existing test cases.
I have added explicit tests.

See below my comment for explanation on why I think this should be the behavior.

Reply:
wait I think I may have misinterpreted some of the existing code.
give me another couple of hours I will verify it.
hopefully I can make this API entirely consistent with others

Reply:
I realize my earlier analysis was incorrect. We can in fact change the behavior of `rolling.count` and `expanding.count` to be consistent with other APIs. Have pushed the code change!